src/main/java/org/dataloader/registries/ScheduledDataLoaderRegistry.java

```diff
@@ -0,0 +1,184 @@
+        this.closed = false;
+    }
+
+    /**
+     * Once closed this registry will never again reschedule checks
+     */
+    @Override
+    public void close() {
+        closed = true;
+    }
+
+    /**
+     * @return how long the {@link ScheduledExecutorService} task will wait before checking the predicate again
+     */
+    public Duration getScheduleDuration() {
+        return schedule;
+    }
+
+    @Override
+    public void dispatchAll() {
+        dispatchAllWithCount();
+    }
+
+    @Override
+    public int dispatchAllWithCount() {
+        int sum = 0;
+        for (Map.Entry<String, DataLoader<?, ?>> entry : dataLoaders.entrySet()) {
+            DataLoader<?, ?> dataLoader = entry.getValue();
+            String key = entry.getKey();
+            if (dispatchPredicate.test(key, dataLoader)) {
+                sum += dataLoader.dispatchWithCounts().getKeysCount();
+            } else {
+                reschedule(key, dataLoader);
+            }
+        }
+        return sum;
+    }
+
+    /**
+     * This will immediately dispatch the {@link DataLoader}s in the registry
+     * without testing the predicate
+     */
+    public void dispatchAllImmediately() {
+        super.dispatchAll();
+    }
+
+    /**
+     * This will immediately dispatch the {@link DataLoader}s in the registry
+     * without testing the predicate
+     *
+     * @return total number of entries that were dispatched from registered {@link org.dataloader.DataLoader}s.
+     */
+    public int dispatchAllWithCountImmediately() {
+        return super.dispatchAllWithCount();
+    }
+
+    /**
+     * This will schedule a task to check the predicate and dispatch if true right now.  It will not do
+     * a pre check of the preodicate like {@link #dispatchAll()} would
+     */
+    public void rescheduleNow() {
+        dataLoaders.forEach(this::reschedule);
+    }
+
+    private void reschedule(String key, DataLoader<?, ?> dataLoader) {
+        if (!closed) {
+            Runnable runThis = () -> dispatchOrReschedule(key, dataLoader);
+            scheduledExecutorService.schedule(runThis, schedule.toMillis(), TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private void dispatchOrReschedule(String key, DataLoader<?, ?> dataLoader) {
+        if (dispatchPredicate.test(key, dataLoader)) {
+            dataLoader.dispatch();
+        } else {
+            reschedule(key, dataLoader);
+        }
+    }
+
+    /**
+     * By default this will create use a {@link Executors#newSingleThreadScheduledExecutor()}
+     * and a schedule duration of 10 milli seconds.
+     *
+     * @return A builder of {@link ScheduledDataLoaderRegistry}s
+     */
+    public static Builder newScheduledRegistry() {
+        return new Builder();
+    }
+
+    public static class Builder {
+
+        private ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+        private DispatchPredicate dispatchPredicate = (key, dl) -> true;
```

Comment:
what is the intent of having the dispatch predicate on the registry vs dataloader?

Reply:
I thought about it on the dataloader itself - but the more I thought about it the more it felt like a cross cutting concern

eg in graphql you dispatch the DataLoaderRegistry and that delegates to each DataLoader

Plus it would mean all the extra baggage of Executors and so on would be in each DataLoader.

This allows for a collection of DataLoaders controlled by the one mechanism.

But since the predicate gets the DL and decides on dispatch or not - you can get as finicky as you like (as if it wa inside the dataloader)